flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/snapshot/RocksIncrementalSnapshotStrategy.java

```diff
@@ -179,6 +179,7 @@ public IncrementalRocksDBSnapshotResources syncPrepareResources(long checkpointI
 
         return new RocksDBIncrementalSnapshotOperation(
                 checkpointId,
+                !checkpointOptions.isForcedFullSnapshot(),
```

Comment:
I think we should add the flag of `forceFullSnapshot` in `SnapshotStrategy#asyncSnapshot` to let user could leverage this information.

Reply:
But we do have it there already in `CheckpointOptions`.